include/bsp/iotc_bsp_tls.h

```diff
@@ -59,188 +58,118 @@ typedef enum iotc_bsp_tls_state_e {
- *     representation.
+ * @param [out] tls_context A pointer to
+ *     {@link ::iotc_bsp_tls_context_t the TLS context}.
  * @param [in] data_ptr A pointer to a buffer to store data that is read.
  * @param [in] data_size The size, in bytes, of the buffer to which data_ptr
  *     points.
  * @param [out] bytes_read The number of bytes read.
- *
- * @retval IOTC_BSP_TLS_STATE_OK All data is successfully read and stored in
- *     the buffer to which data_ptr points.
- * @retval IOTC_BSP_TLS_STATE_WANT_READ The TLS handshake is partially complete.
- *     The function will be called again to read the remaining data from the
- *     socket specified in the tls_context parameter.
- * @retval IOTC_BSP_TLS_STATE_READ_ERROR Can't read data.
  */
 iotc_bsp_tls_state_t iotc_bsp_tls_read(iotc_bsp_tls_context_t* tls_context,
                                        uint8_t* data_ptr, size_t data_size,
                                        int* bytes_read);
 
 /**
- * @function
- * @brief Count the pending readable bytes.
- *
- * Return the the number of bytes that are available to be
- * read by the <code>iotc_bsp_tls_read</code> implementation.
- *
- * @see iotc_bsp_tls_context_t
+ * @brief Gets the pending readable bytes.
  *
- * @param [in] tls_context The iotc_bsp_tls_context_t TLS context
- *     representation.
- *
- * @return The number of bytes available to read.
+ * @param [out] tls_context A pointer to
+ *     {@link ::iotc_bsp_tls_context_t the TLS context}.
  */
 int iotc_bsp_tls_pending(iotc_bsp_tls_context_t* tls_context);
 
 /**
- * @function
  * @brief Write data to a socket.
  *
- * @see iotc_bsp_tls_context_t
- *
- * @param [in] tls_context The iotc_bsp_tls_context_t TLS context
- *     representation.
+ * @param [out] tls_context A pointer to
+ *     {@link ::iotc_bsp_tls_context_t the TLS context}.
  * @param [in] data_ptr A pointer to a buffer with the data to be sent.
- * @param [in] data_size the size, in bytes, of the buffer to which data_ptr
- *     points.
+ * @param [in] data_size The size, in bytes, of the buffer to which
+ *     {@code data_ptr} points.
  * @param [out] bytes_written The number of bytes written.
- *
- * @retval IOTC_BSP_TLS_STATE_OK All data is successfully written.
- * @retval IOTC_BSP_TLS_STATE_WANT_WRITE The TLS handshake is partially
- *     complete.The function will be called again to write the remaining
- *     data to the socket specified in the tls_context parameter.
- * @retval IOTC_BSP_TLS_STATE_WRITE_ERROR Can't write data.
  */
 iotc_bsp_tls_state_t iotc_bsp_tls_write(iotc_bsp_tls_context_t* tls_context,
                                         uint8_t* data_ptr, size_t data_size,
                                         int* bytes_written);
 
-/** Notify the client application to read data on a socket. Don't modify this
- * function. */
+/** @details Notifies the client application to read data on a socket. Don't
+ * modify this function. */
```

Comment:
"... Implemented inside the SDK, don't modify this function."

Reply:
Done